lib/maths/CTimeSeriesTestForSeasonality.cc

```diff
@@ -374,6 +376,7 @@ CSeasonalDecomposition CTimeSeriesTestForSeasonality::decompose() const {
                     trend.add(static_cast<double>(j), CBasicStatistics::mean(values[j]),
                               CBasicStatistics::count(values[j]));
                 }
+                trend.parameters(parameters);
```

Comment:
I am probably mistaken, but it seems to me that `parameters` is getting set but not read from here on. Why do we need to store it if we don't use the value?

Reply:
This is correct, but the parameters are referenced by the predictor. This is actually a significant performance optimisation since the runtime is dominated by computing these parameters, I'll add a comment to explain what is going on here.

Reply:
See comment in [079a076](https://github.com/elastic/ml-cpp/pull/1550/commits/079a07614ed72f3b9562f478b634eaedbcd68f77).